api/views/teacher.py

```diff
@@ -123,6 +124,29 @@ def expelled(  # noqa: ARG002
         TeacherProcessor.expelled(teacher)
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    @extend_schema(
+        responses={
+            status.HTTP_204_NO_CONTENT: OpenApiResponse(description="Action is taken"),
+            status.HTTP_400_BAD_REQUEST: OpenApiResponse(
+                response=ValidationErrorSerializer,
+                description="Something is wrong with the query params",
+            ),
+            status.HTTP_422_UNPROCESSABLE_ENTITY: OpenApiResponse(
+                response=BaseAPIExceptionSerializer,
+                description="Unable to process teacher with group(s)",
```

Comment:
Actually, a ConlifError seems good to me here too

Reply:
> Actually, a ConlifError seems good to me here too

Maybe you're right. `422` signalizes that the client should not send the request with the same data again. But actually, if the client removes the groups in some other request, then they can actually resend the request here with the same teacher data - and succeed.